ethereum_history_api/circuits/lib/src/rlp/types.nr

```diff
@@ -25,26 +25,61 @@ struct RlpFragment {
 }
 
 impl RlpFragment {
-    fn assert_eq_bytes<FIELD_NAME_LEN, MAX_RLP_LEN, MAX_VALUE_LEN>(
+    fn assert_eq_bytes_length<FIELD_NAME_LEN, MAX_RLP_LEN, MAX_VALUE_LEN>(
         self,
         field_name: str<FIELD_NAME_LEN>,
         rlp: [u8; MAX_RLP_LEN],
-        value: BoundedVec<u8, MAX_VALUE_LEN>
+        value: [u8; MAX_VALUE_LEN],
+        length: u64
     ) {
         assert(self.data_type == STRING, f"{field_name}: Invalid RLP type");
-        assert(self.length == value.len, f"{field_name}: Invalid RLP length");
+        assert(self.length == length, f"{field_name}: Invalid RLP length");
         assert(
-            sub_array_equals_up_to_length(value.storage, rlp, self.offset, self.length), f"{field_name}: Invalid RLP value"
+            sub_array_equals_up_to_length(value, rlp, self.offset, self.length), f"{field_name}: Invalid RLP value"
         );
     }
 
-    fn assert_eq_string<FIELD_NAME_LEN, MAX_RLP_LEN, MAX_VALUE_LEN>(
+    fn assert_eq_bounded_vec<FIELD_NAME_LEN, MAX_RLP_LEN, MAX_VALUE_LEN>(
         self,
         field_name: str<FIELD_NAME_LEN>,
         rlp: [u8; MAX_RLP_LEN],
         value: BoundedVec<u8, MAX_VALUE_LEN>
     ) {
-        self.assert_eq_bytes(field_name, rlp, value);
+        self.assert_eq_bytes_length(field_name, rlp, value.storage, value.len);
+    }
+
+    fn assert_eq_null<FIELD_NAME_LEN, MAX_RLP_LEN>(self, field_name: str<FIELD_NAME_LEN>) {
+        assert(self.data_type == STRING, f"{field_name}: Invalid RLP type");
```

Comment:
As Marek suggested in one of the comments, I will change assert message here for "Expected empty string" (it gives more information in case of the failure). Then it is clearer that we shouldn't reuse `assert_eq_bounded_vec`